src/node/setupServer.ts

```diff
@@ -1,15 +1,137 @@
+      'server-setup',
+      handlers,
+    )
+
+    this.resolvedOptions = {} as RequiredDeep<SharedOptions>
+
+    this.init()
+  }
+
+  /**
+   * Subscribe to all requests that are using the interceptor object
+   */
+  public async init(): Promise<void> {
+    this.interceptor.on('request', async (request) => {
+      const mockedRequest = new MockedRequest(request.url, {
+        ...request,
+        body: await request.arrayBuffer(),
+      })
+
+      const response = await handleRequest<
+        MockedInterceptedResponse & { delay?: number }
+      >(
+        mockedRequest,
+        this.currentHandlers,
+        this.resolvedOptions,
+        this.emitter,
+        {
+          transformResponse(response) {
+            return {
+              status: response.status,
+              statusText: response.statusText,
+              headers: response.headers.all(),
+              body: response.body,
+              delay: response.delay,
+            }
+          },
+        },
+      )
+
+      if (response) {
+        // Delay Node.js responses in the listener so that
+        // the response lookup logic is not concerned with responding
+        // in any way. The same delay is implemented in the worker.
+        if (response.delay) {
+          await new Promise((resolve) => {
+            setTimeout(resolve, response.delay)
+          })
+        }
+
+        request.respondWith(response)
+      }
+
+      return
+    })
+
+    this.interceptor.on('response', (request, response) => {
+      if (!request.id) {
+        return
+      }
+
+      if (response.headers.get('x-powered-by') === 'msw') {
+        this.emitter.emit('response:mocked', response, request.id)
+      } else {
+        this.emitter.emit('response:bypass', response, request.id)
+      }
+    })
+  }
+
+  public listen(options: Record<string, any> = {}): void {
+    this.resolvedOptions = mergeRight(
+      DEFAULT_LISTEN_OPTIONS,
+      options,
+    ) as RequiredDeep<SharedOptions>
+    super.apply()
+  }
+
+  public printHandlers() {
+    const handlers = this.listHandlers()
+
+    handlers.forEach((handler) => {
+      const { header, callFrame } = handler.info
+
+      const pragma = handler.info.hasOwnProperty('operationType')
+        ? '[graphql]'
+        : '[rest]'
+
+      console.log(`\
+${bold(`${pragma} ${header}`)}
+Declaration: ${callFrame}
+`)
+    })
+  }
+
+  public close(): void {
+    super.dispose()
+  }
+}
 
 /**
  * Sets up a requests interception in Node.js with the given request handlers.
  * @param {RequestHandler[]} requestHandlers List of request handlers.
  * @see {@link https://mswjs.io/docs/api/setup-server `setupServer`}
  */
-export const setupServer = createSetupServer(
-  // List each interceptor separately instead of using the "node" preset
-  // so that MSW wouldn't bundle the unnecessary classes (i.e. "SocketPolyfill").
-  ClientRequestInterceptor,
-  XMLHttpRequestInterceptor,
-)
```

Comment:
The `createSetupServer` function is not being used for the node server anymore, but it's still being used for `src/native` so I didn't delete the file.

Reply:
We can leave it for now, good call.